cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CruiseControlReconciler.java

```diff
@@ -247,52 +246,74 @@ protected Future<Void> certificatesSecret(Clock clock) {
-     * 
-     * If the Topic Operator component is not enabled and Cruise Control's secret is present, we reuse the old Cruise Control secret's content. 
-     * If the Topic Operator component is not enabled and Cruise Control's secret is not present, we generate a new Cruise Control secret.
-     * 
-     * In any case, we generate the Cruise Control secret's content hash, that is later used to detect changes that require a pod restart.
+     *  (3) Topic Operator-managed API secret
+     *      The Topic Operator-managed API secret contains a username and password for the Topic Operator user, which
+     *      has ADMIN role by default.
+     *
+     * If there are any changes to the API user content in any of the secrets, the centralized Cruise Control secret is
+     * updated with the new changes. Otherwise the content in the centralized Cruise Control secret remains the same.
+     *
+     * Strimzi maintains a hash of the centralized Cruise Control API secret content to detect changes that require a pod restart.
      *
      * @return Future which completes when the reconciliation is done.
      */
-    protected Future<Void> apiSecret() {
+    protected Future<Void> apiSecrets() {
         if (cruiseControl != null) {
+
+            Integer centralizedApiSecretIndex = 0;
+            Integer userManagedApiSecretIndex;
+            Integer topicOperatorManagedApiSecretIndex;
+
+            List<Future<Secret>> futures = new ArrayList<>();
+            // Centralized API secret
+            futures.add(secretOperator.getAsync(reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name())));
+
+            if (cruiseControl.apiUsersEnabled()) {
+                // User-managed API secret
+                futures.add(secretOperator.getAsync(reconciliation.namespace(), cruiseControl.getUserManagedApiSecretName()));
+                userManagedApiSecretIndex = futures.size() - 1;
+            } else {
+                userManagedApiSecretIndex = null;
+            }
+
             if (isTopicOperatorEnabled) {
-                return Future.join(
-                    secretOperator.getAsync(reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name())),
-                    secretOperator.getAsync(reconciliation.namespace(), KafkaResources.entityTopicOperatorCcApiSecretName(reconciliation.name()))
-                ).compose(
+                // Topic operator-managed API secret
+                futures.add(secretOperator.getAsync(reconciliation.namespace(), KafkaResources.entityTopicOperatorCcApiSecretName(reconciliation.name())));
+                topicOperatorManagedApiSecretIndex = futures.size() - 1;
+            } else {
+                topicOperatorManagedApiSecretIndex = null;
+            }
```

Comment:
This seems weird. Maybe instead of the indexes, you can just use succeeded futures with null result. And if you name them instead of using list, you save the list creation.

Reply:
You are right, just cleaned it up now